derp/derphttp/mesh_client.go

```diff
@@ -112,7 +116,21 @@ func (c *Client) RunWatchConnectionLoop(ctx context.Context, ignoreServerKey key
 	}
 
 	for ctx.Err() == nil {
-		if c.ServerPublicKey() == ignoreServerKey {
+		// Make sure we're connected before calling s.ServerPublicKey.
+		_, _, err := c.connect(ctx, "RunWatchConnectionLoop")
```

Comment:
I guess we want to do this every time we get here because once it succeeds, we only come back here by getting an error in RecvDetail below? (Which, I guess, probably means we have lost our connection?)

Reply:
Yup. Either initially (when no connection) or on loops around which means it had broke.